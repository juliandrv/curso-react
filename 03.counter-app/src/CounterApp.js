import { useState } from "react";
import { PropTypes } from "prop-types";

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  // handleAdd
  const handleAdd = () => {
    // setCounter((c) => c + 1);
    setCounter(counter + 1);
  }

  // handleSubstract
  const handleSubstract = () => {
    setCounter(counter - 1);
  }

  // handleReset
  const handleReset = () => {
    setCounter(value);
  }

  return (
    <>
      <h1>CounterApp</h1>
      <div>
        <button onClick={handleSubstract}>- 1</button>
        <h2> {counter} </h2>
        <button onClick={handleAdd}>+ 1</button>
      </div>
      <button onClick={handleReset}>Reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp;

