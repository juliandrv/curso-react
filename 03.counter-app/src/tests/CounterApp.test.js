import '../setUpTests'
import { shallow } from "enzyme";
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('debe mostrar <CounterApp /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar el valor por defecto de 100', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const counter = wrapper.find('h2').text().trim();
    expect(counter).toBe(`${value}`)
  });

  test('debe decrementar el contador con el botón -1', () => {
    wrapper.find('button').at(0).simulate('click', {});
    const counter = wrapper.find('h2').text().trim();
    expect(counter).toBe('9')
  });

  test('debe incrementar el contador con el botón +1', () => {
    wrapper.find('button').at(1).simulate('click', {});
    const counter = wrapper.find('h2').text().trim();
    expect(counter).toBe('11')
  });

  test('debe reestablecer el contador a 10', () => {
    const wrapper = shallow(<CounterApp value={100} />);
    wrapper.find('button').at(1).simulate('click', {});
    wrapper.find('button').at(1).simulate('click', {});
    wrapper.find('button').at(2).simulate('click', {});
    const counter = wrapper.find('h2').text().trim();
    expect(counter).toBe('100')
  });

})