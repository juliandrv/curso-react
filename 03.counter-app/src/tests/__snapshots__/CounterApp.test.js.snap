// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <div>
    <button
      onClick={[Function]}
    >
      - 1
    </button>
    <h2>
       
      10
       
    </h2>
    <button
      onClick={[Function]}
    >
      + 1
    </button>
  </div>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
